src/compiler/compile-props.js

```diff
@@ -344,6 +331,36 @@ export function coerceProp (prop, value) {
   return coerce(value)
 }
 
+function assertType (value, type) {
```

Comment:
Let's keep it here for now since it's not yet used in any other context. We can move it when we do need to use it elsewhere.